backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Linker.kt

```diff
@@ -18,6 +19,13 @@ internal fun determineLinkerOutput(context: Context): LinkerOutputKind =
             else -> TODO("${context.config.produce} should not reach native linker stage")
         }
 
+// TODO: Move to a more appropriate place.
+internal val KonanTarget.isAppleTarget: Boolean
```

Comment:
This name doesn't reflect the actual conditions that need to be checked on its use sites.
This is generally more like `IsThisParticularKindOfObjCInteropEnabled`.
I would prefer using old target-family-based approach for now.

Reply:
Unless you have short-term plans to add more relevant target families of course.

Reply:
Yes, I think we should introduce `Family.TVOS`, `Family.WATCHOS` because different flavours of `iOS` have different platform libs and runtime dependencies. And `when` over all Apple targets is getting kinda cumbersome.

Reply:
You mean platform libraries are deduced from the family in https://github.com/JetBrains/kotlin-native/blob/bcfd196c0498195e3eecd7ef4e91fa2c7f6a194f/platformLibs/build.gradle#L39? This could be worked around (like hack for MIPS zlib), but indeed may be a good idea to use OS family as identity for the set of platform libs.

Reply:
Yes, that's what I mean.